src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -1861,19 +1861,25 @@ def listAllCMSSWConfigCacheIDs(self):
 
     def setTrustLocationFlag(self, inputFlag=False, pileupFlag=False):
         """
-        _setTrustLocationFlag_
-
         Set the input and the pileup flags in the top level tasks
         indicating that site lists should be used as location data
 
         The input data flag has to be set only for top level tasks, otherwise
         it affects where secondary jobs are meant to run.
         The pileup flag has to be set for all the tasks in the workload.
+
+        Validate these parameters to make sure they are only set for workflows
+        that require those type of input datasets (ACDCs are not validated)
         """
-        if inputFlag is True and not self.listInputDatasets():
+        isACDCWorkflow = False
+        for task in self.taskIterator():
+            if task.getInputACDC():
+                isACDCWorkflow = True
+
+        if inputFlag is True and isACDCWorkflow is False and not self.listInputDatasets():
             msg = "Setting TrustSitelists=True for workflows without input dataset is forbidden!"
             raise RuntimeError(msg)
-        if pileupFlag is True and not self.listPileupDatasets():
+        if pileupFlag is True and isACDCWorkflow is False and not self.listPileupDatasets():
```

Comment:
The full permutation of those three flags is not exhausted with those two checks (this one here and the one from above). Have you checked if the other cases are relevant to be checked too?

Reply:
I can only think about those 2 use cases that should fail:
1. input flag is enabled, the workflow is NOT an ACDC and there is no input dataset --> fail assignment
2. pileup flag is enabled, the workflow is NOT an ACDC and there is no pileup dataset --> fail assignment

If flags are false, go ahead.
If workflow is an ACDC, go ahead.
If workflow has input dataset, go ahead.
If workflow has pileup dataset, go ahead.